pkg/secrets/in_memory.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"context"
+	"fmt"
+)
+
+// Compile-time check to verify implements interface.
+var _ SecretManager = (*InMemory)(nil)
+
+// InMemory is an in-memory secret manager, primarily used for testing.
+type InMemory struct {
```

Comment:
do you want to add the ability to add new secrets after create?
I did this for the InMemory KMS

Reply:
I did that originally, but it requires a type assertion since we return an interface. I added the  `NewInMemoryFromMap` so you could create one with values preset. I'm open to allowing it to mutate state, just didn't see a use case for it.